packages/sdk-cli/src/models/localRelay/index.ts

```diff
@@ -29,32 +41,19 @@ export async function instance(): Promise<RelayInfo> {
   }
 
   const relayJsPath = await relayEntryPointPath();
+  const logStream = await createLogStream(RELAY_LOG_FILE_PATH);
+  const relayProcess = await launch([relayJsPath], logStream);
 
-  const logStream = createWriteStream(RELAY_LOG_FILE_PATH);
+  relayProcess.on('error', (error: Error) => {
+    console.error('Local relay process threw error:', error);
+    // 'error' doesn't guarantee that the process exits afterwards
+    relayProcess.kill('SIGKILL');
 
-  // https://stackoverflow.com/a/44846808/6539857
-  // Without 'open' event spawn() won't accept the WriteStream, because
-  // "[log stream] must have an underlying descriptor (file streams do not until the 'open' event has occurred)"
-  // Related: https://github.com/nodejs/node-v0.x-archive/issues/4030
-  logStream.on('open', (fd: number) => {
-    const relayProcess = launch([relayJsPath], logStream);
-
-    relayProcess.on('error', (error: Error) => {
-      console.error('Local relay process threw error:', error);
-      relayProcess.kill('SIGKILL');
-      throw error;
-    });
-
-    relayProcess.on('close', async () => {
-      console.warn('Local relay process exited and closed');
-    });
+    throw error;
```

Comment:
Again, I'm not sure throwing inside an error handler actually does anything? It might re-raise the error handler perhaps, but given you're already in it?

Reply:
I've tested this and it seems to be like it behaves just like any other error throw, i.e. it will just abruptly end the process. I'm not very sure what the process is exactly though...is it only the CLI shell that is spawned by `npx fitbit` or is it `npx fitbit`, too?

Reply:
It'll maybe end the process because Node sees an uncaught error, but that's not a particularly clean way of exiting. `npx fitbit` _is_ the CLI shell (sdk-cli Vorpal shell)

Reply:
Log error and `process.exit(negative_number)` then?

Reply:
Move this code into `launch.ts` and add the rejection case suggested there, at which point it'll be thown via the await at line 45.

Reply:
`'error'` can be emitted at any point in future though; if I move it to `launch` and add the event listener inside the Promise, the Promise will reject **only** if `'error'` is emitted on *start*, which is only 1 of the possible `'error'` emission cases: https://nodejs.org/api/child_process.html#event-error

Reply:
But do the other two cases apply? For 2. I think you're only killing in this error case and 3. you aren't using pipes to communicate.

Reply:
Hmm right. That certainly makes things easier, thanks

Reply:
https://github.com/Fitbit/developer-bridge/pull/176/commits/680048b82c1fae3414b669d2fd23a4ced3669690